how-to/postgrest.md

```diff
@@ -397,4 +397,168 @@ Now, you can perform a keyword-based search directly from your web browser. Simp
 ]
 ```
 
-This seamless and user-friendly approach to searching through your data with keywords is both practical and exciting. It's a fantastic way to enhance your search capabilities, don't you think?
\ No newline at end of file
+This seamless and user-friendly approach to searching through your data with keywords is both practical and exciting. It's a fantastic way to enhance your search capabilities, don't you think?
+
+
+## Using Keyword Auto Complete
+
+PGroonga has features to implement auto complete which is explained in [this how to section](https://pgroonga.github.io/how-to/auto-complete.html).
+
+Here we will explor how to implement this using PostgREST and just a simple html with JavaScript.
+
+### Create Table for Auto Complete Feature
+
+```sql
+CREATE TABLE terms (
+  term text,
+  readings text[]
+);
+
+CREATE INDEX pgroonga_terms_prefix_search ON terms USING pgroonga
+  (readings pgroonga_text_array_term_search_ops_v2);
+
+CREATE INDEX pgroonga_terms_full_text_search ON terms USING pgroonga
+  (term)
+  WITH (tokenizer = 'TokenBigramSplitSymbolAlphaDigit');
+
+INSERT INTO terms (term, readings) VALUES ('牛乳', ARRAY['ギュウニュウ', 'ミルク']);
+INSERT INTO terms (term, readings) VALUES ('PostgreSQL', ARRAY['ポスグレ', 'posugure', 'postgres']);
+INSERT INTO terms (term, readings) VALUES ('Groonga', ARRAY['guru-nga', 'グルンガ','ぐるんが','gurunga']);
+INSERT INTO terms (term, readings) VALUES ('PGroonga', ARRAY['pi-ji-runga', 'ピージールンガ','ぴーじーるんが','pg','ピージー']);
+```
+
+### Set Up PostgREST Permission
+
+```sql
+RANT SELECT ON terms TO web_user;
+```
+
+### Create Auto Complete End Point
+
+```sql
+CREATE OR REPLACE FUNCTION autocomplete(keyword text) RETURNS SETOF text AS $$
+DECLARE
+  result text[];
+BEGIN
+  IF keyword = '' THEN
+    RETURN QUERY SELECT unnest(result);
+  ELSE
+    RETURN QUERY SELECT term FROM terms WHERE readings &^~ keyword;
+  END IF;
+END;
+$$ LANGUAGE plpgsql;
+```
+
+### Create a html with JavaScript
+
+Create following HTML file:
+
+```sh
+vi index.html
+```
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Groonga dictionary search</title>
```

Comment:
Could you update the title?

Reply:
Changed to "PGroonga Auto Complete Search"